### PR TITLE
Default path for home folder in user_files.sls if home omitted in pillar.

### PR DESCRIPTION
**Pull request by hatifnatt**
Fix for #99 
Rendering SLS 'base:users.user_files' failed: Jinja variable 'dict object' has no attribute 'home'

### CODE REVIEW

**Comment by ketzacoatl** [comment]
This is an obvious win, we should be including sensible defaults like this.

One update I would recommend to the PR: use a `{% set home = ...%}` here: https://github.com/hatifnatt/users-formula/blob/2763338abe4b3dc443734ce2cde06e600b557e09/users/user_files.sls#L9, and then use that variable in `- home: {{ home }}` on L31. This style/pattern is already in use in the formula, and in general, makes for _much_ more readable formula - I have come to appreciate this pattern as my formula grow and get full of more text.

**Comment by 0xf10e** [comment]
Hmmm, the parent directory for user's $HOME should use the OS' configured value if not set explicitly via pillar.
On FreeBSD running `useradd` without any parameters returns the current defaults.
I *think* on Ubuntu you have to scrape the defaults from `/etc/default/useradd` (see [useradd(8)](http://manpages.ubuntu.com/manpages/trusty/man8/useradd.8.html)).

**Comment by hatifnatt** [comment]
@0xf10e thats sound reasonable, so we need to set up defaults via map.jinja? It is already used for some other system specific values. Unfortunately my experience in Salt not enough, and I'm not sure how properly implement this. Also I doesn't have any FreeBSD systems to make some tests. Currently I only have Debian.
What I can do - make changes in style as @ketzacoatl recommended above. But this only can be temporary workaround until implementation which cares about system specific difference.

**Comment by 0xf10e** [comment]
I guess I can wrap the commands we need for this "inline" (in the statefile) but a cleaner way would be adding a grain for this.

My first thought was about FreeBSD but I also got some hosts where `/home` will be an NFS mountpoint so the few local homedirs should go somewhere else like `/usr/home` (even on Ubuntu).

Oh yeah, and the `/usr/home` vs `/home` thing seems to be a [bit more complicated](https://twitter.com/0xf10e/status/664802653227122688) than I thought as it depends on installation time decisions ;)

**Comment by ketzacoatl** [comment]
@hatifnatt, yes, you would add that in `map.jinja`. That map is going to emulate / serve as a map for all defaults for all systems.. and unfortunately, we would be hardcoding those defaults into that map (rather than getting them from the system we're on), but other details make that necessary.

**Comment by aboe76** [comment]
So can this be merged or refactored according to @ketzacoatl style? this has been sitting here rather long...

File: users/user_files.sls [hunk 1]
```diff
@@ -28,7 +28,7 @@ include:
 {%- if not skip_user %}
 users_userfiles_{{ username }}_recursive:
   file.recurse:
-    - name: {{ user.home }}
+    - name: {{ salt['pillar.get']( username ~ 'user:home', '/home/' ~ username )}}
```
Comment (babilen) [inline users/user_files.sls]: Shouldn't that be `salt['pillar.get']('users:' ~ username ~ ':home', '/home/' ~ username )` ?

**Comment by hatifnatt** [comment]
I have made "style" changes a long time ago https://github.com/hatifnatt/users-formula/blob/hatifnatt-user_files.sls/users/user_files.sls but after reading @0xf10e comments about base '/home' folder location I decided that my changes not enough universal to commit them.

**Comment by aboe76** [comment]
@hatifnatt your implementations looks the cleanest in my opinion,

**Comment by hatifnatt** [comment]
@aboe76 I think I need test it again, and if it work correctly I will make new pull request.

File: users/user_files.sls (hunk 1 above)
Comment (hatifnatt) [inline users/user_files.sls]: Sure I used exactly same code in my "refactored" branch https://github.com/hatifnatt/users-formula/blob/hatifnatt-user_files.sls/users/user_files.sls#L10

**Comment by hatifnatt** [comment]
I tested my code once again, seems it work fine so I created new pull request https://github.com/saltstack-formulas/users-formula/pull/114
I think this one need to be closed now?

**Comment by hatifnatt** [comment]
I tested my code once again, seems it work fine so I created new pull request https://github.com/saltstack-formulas/users-formula/pull/114
I think this one need to be closed now?

**Comment by aboe76** [comment]
Ok will close this one and merge the other one,

**Pull request closed without merging**